### ISSUE TITLE
Retriving related record with «null» value

### ISSUE THREAD

**Issue by dugwood** [phalcon/cphalcon#2953]
The model should not request a related object which value is null.

Say we have:
```php
/**
 * @Source('domains')
 * @BelongsTo('masterId', 'Domain', 'id', {alias="master"})
 */
class Domain extends Model
{
	/**
	 * @var integer $id
	 *
	 * @Primary
	 * @Identity
	 * @Column(column="id", type="integer")
	 */
	public $id;

	/**
	 * @var string $masterId
	 *
	 * @Column(column="master", type="integer")
	 */
	public $masterId;
}

$domain = new Domain(1); // SELECT * FROM domains WHERE id = 1
echo $domain->master; // SELECT * FROM domains WHERE id = null
```

If domains.master is null in the database, there should not be a query when accessing $domain->master.

AFAIK, no object can be indexed with NULL value. Of course 0 (zero) is a valid value, so it must be requested in database.

Note: it may need {foreignKey="error message"} in the BelongsTo definition, but for now the NULL value isn't allowed as a valid foreignKey (see https://github.com/phalcon/cphalcon/issues/2039)

**Comment by andresgutierrez** [comment]
NULLs in foreign keys implemented in 2.0.3:

```php
class RobotsParts extends Model
{
    public function initialize()
    {
        $this->belongsTo("robots_id", "Some\Robots", "id", array(
            "foreignKey" => [
                "message" => "Robots is required",
                "allowNulls" => true
            ]
        ));
    }
}
```

**Comment by dugwood** [comment]
@andresgutierrez Thanks for the fix.

Did you check the first part of the issue too? I mean never request a «id = NULL» on the database? I've fixed it in the PHP Model class inherited from Phalcon\Mvc\Model, but that would be great to remove this PHP code of mine:
```
	public static function findFirst($parameters = null)
	{
		/* Foreign key is NULL, we shouldn't request it against the database (https://github.com/phalcon/cphalcon/issues/2953) */
		if (is_array($parameters) && isset($parameters[0]) && array_key_exists(0, $parameters['bind']) && is_null($parameters['bind'][0]))
		{
			return false;
		}
```
